src/frontend/context/SecurityContext.tsx

```diff
@@ -0,0 +1,128 @@
+import * as React from "react";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import { ASYNC_KILL_KEY, PASSWORD_KEY } from "../constants";
+import { useEffect } from "react";
+import { useRef } from "react";
+
+//We should be setting the password to null when the password is turned off
+interface SecuritContextType {
+  killState: boolean;
+  setKillState: (killStateValue: boolean) => void;
+  passcode: string | null;
+  setPasscode: (passValue: string | null) => void;
+  checkFlag: boolean;
+  killStateActive: boolean;
+  setKillStateActive: (isActive: boolean) => void;
+  setCheckFlag: (checkFlag: boolean) => void;
+}
+
+const DEFAULT_CONTEXT: SecuritContextType = {
+  killState: false,
+  setKillState: () => {},
+  setPasscode: () => {},
+  passcode: null,
+  checkFlag: false,
+  setCheckFlag: () => {},
+  killStateActive: false,
+  setKillStateActive: () => {},
+};
+
+export const SecurityContext = React.createContext<SecuritContextType>(
+  DEFAULT_CONTEXT
+);
+
+export const SecurityProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const [killState, setKillState] = React.useState(false);
+  const [killStateActive, setKillStateActive] = React.useState(false);
+  const [passcode, setPasscode] = React.useState<string | null>(null);
+  const [checkFlag, setCheckFlag] = React.useState(false);
+  const firstRender = useRef(true);
+
+  useEffect(() => {
+    if (firstRender.current) {
+      AsyncStorage.getItem(PASSWORD_KEY, (err, val) => {
+        if (!!val) {
+          setPasscode(val);
+          setCheckFlag(true);
+          AsyncStorage.getItem(ASYNC_KILL_KEY, (err, val) => {
+            if (!!val) setKillStateActive(val === "true");
+          });
+        } else {
+          firstRender.current = false;
+        }
+      });
```

Comment:
prefer using the promise-based API, as the callback style is considered legacy https://react-native-async-storage.github.io/async-storage/docs/api#getitem

Reply:
also, consider using [AsyncStorage.multiget](https://react-native-async-storage.github.io/async-storage/docs/api#multiget) instead e.g.

```TS
AsyncStorage.multiGet([PASSWORD_KEY, ASYNC_KILL_KEY])
  .then(([password, killState]) => { ... })
  .catch(err => { ... })
```